src/plugins/explore/public/application/utils/languages/ppl/get_query_string_with_source/get_query_with_source.ts

```diff
@@ -16,12 +16,26 @@ export const getQueryWithSource = (query: Query): QueryWithQueryAsString => {
   const hasDescribe = /^\s*describe\s+/.test(lowerCaseQuery);
   const hasShow = /^\s*show\s+/.test(lowerCaseQuery);
 
+  // Temporarily adding backticks to dataset type INDEXES or INDEX_PATTERNS to until these two issues are resolved:
+  // https://github.com/opensearch-project/sql/issues/4444
+  // https://github.com/opensearch-project/sql/issues/4445
+  let datasetTitle: string;
+  if (query.dataset && ['INDEXES', 'INDEX_PATTERN'].includes(query.dataset.type)) {
+    if (hasSource) {
+      // Replace source=anything with source=`anything` (handling spaces around = and ensuring anything is not already backticked)
+      const updatedQuery = queryString.replace(/(\bsource\s*=\s*)([^`\s][^\s]*)/gi, '$1`$2`');
```

Comment:
`[^\s]` is just `\S`? and it includes pipes, does it work if query is `source=index|where 1=1`?

Reply:
<img width="403" height="147" alt="Screenshot 2025-10-07 at 9 25 59 AM" src="https://github.com/user-attachments/assets/71892419-ae25-46f7-82b9-692d045b2fd2" />

yeah my IDE shows that too.

And no it doesn't solve your scenario, good catch, let me put up a fix